### PR TITLE
Docs update for Deconvolution2D layer

### PR DESCRIPTION
**Pull request by lukovkin**
Implemented Deconvolution2D layer was missing documentation, which is important especially because it's required to infer output shape for this layer explicitly and pass as a parameter.
One of the raised issues - #3540 

Added standard docs, references and guidelines to the output shape calculation.
Also fixed typo in the exception message.

### CODE REVIEW

**Comment by lukovkin** [comment]
Err... The Travis build failure seems to be unrelated to the changes, it fails Python 3.4 run on test_sequential_model_saving(): https://travis-ci.org/fchollet/keras/jobs/154395874#L1294-L1300

File: keras/layers/convolutional.py [hunk 1]
```diff
@@ -382,10 +382,94 @@ def get_config(self):
...
+                    a - user-specified quantity used to distinguish between
+                        the s different possible output sizes.
+        init: name of initialization function for the weights of the layer
+            (see [initializations](../initializations.md)), or alternatively,
+            Theano function to use for weights initialization.
+            This parameter is only relevant if you don't pass
+            a `weights` argument.
+        activation: name of activation function to use
+            (see [activations](../activations.md)),
+            or alternatively, elementwise Theano function.
```
Comment (fchollet) [inline keras/layers/convolutional.py]: Theano -> Theano/TensorFlow

File: keras/layers/convolutional.py [hunk 2]
```diff
@@ -382,10 +382,94 @@ def get_config(self):
...
+            (eg. maxnorm, nonneg), applied to the main weights matrix.
+        b_constraint: instance of the [constraints](../constraints.md) module,
+            applied to the bias.
+        dim_ordering: 'th' or 'tf'. In 'th' mode, the channels dimension
+            (the depth) is at index 1, in 'tf' mode is it at index 3.
+            It defaults to the `image_dim_ordering` value found in your
+            Keras config file at `~/.keras/keras.json`.
+            If you never set it, then it will be "th".
+        bias: whether to include a bias (i.e. make the layer affine rather than linear).
+    # Input shape
```
Comment (fchollet) [inline keras/layers/convolutional.py]: Blank line required before the next section

File: keras/layers/convolutional.py [hunk 3]
```diff
@@ -382,10 +382,94 @@ def get_config(self):
...
+        4D tensor with shape:
+        `(samples, channels, rows, cols)` if dim_ordering='th'
+        or 4D tensor with shape:
+        `(samples, rows, cols, channels)` if dim_ordering='tf'.
+    # Output shape
```
Comment (fchollet) [inline keras/layers/convolutional.py]: Blank line required before the next section

File: keras/layers/convolutional.py [hunk 4]
```diff
@@ -382,10 +382,94 @@ def get_config(self):
...
+        4D tensor with shape:
+        `(samples, nb_filter, new_rows, new_cols)` if dim_ordering='th'
+        or 4D tensor with shape:
+        `(samples, new_rows, new_cols, nb_filter)` if dim_ordering='tf'.
+        `rows` and `cols` values might have changed due to padding.
+
+    # References
+        [1] [A guide to convolution arithmetic for deep learning] arXiv:1603.07285v1 [stat.ML]
```
Comment (fchollet) [inline keras/layers/convolutional.py]: First 2 references should be formatted as markdown links

**Comment by fchollet** [comment]
LGTM

**Pull request merged**